test/zdtm.py

```diff
@@ -1156,7 +1168,10 @@ def __criu_act(self, action, opts=[], log=None, nowait=False):
         if action == "restore":
             preexec = None
         else:
-            preexec = self.__user and self.set_user_id or None
+            if os.getuid():
+                preexec = None
+            else:
+                preexec = self.__user and self.set_user_id or None
```

Comment:
hmm, does the --user option do what we need?

Reply:
I took a quick look. The code that currently exists around this option is pretty minimal and was introduced in dde2adbc9d5a988a3624ab1056d43cef6cfd58c0.

It removes `suid` and `nouser` tests and `ns` and `uns` flavours from the launcher config and runs only `criu dump` as non-root, but that's not sufficient. Dumps fail when trying to read `/proc/$process/map_files`.

`--rootless` drops to user before invoking `zdtm_ct`.

It seems like `--user` hasn't been exercised in a while and is broken, unless I'm missing something.

Reply:
I think we had the same discussion in my original PR with the same result that `--user` is a bit different. It is used in one of the Jenkins script, although I am not sure if that script is actually used by Jenkins.